File: mono/src/symbol/method_map.rs
use serde::{Serialize, Deserialize};

use crate::symbol::{Declaration, Function, Property};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MethodMap {
    #[serde(flatten)]
    pub declaration: Declaration,

    /// Parent inheritance if any
    pub parent: Option<String>,

    /// Functions within this methodmap
    pub methods: Vec<Function>,

    /// Properties within this methodmap
    pub properties: Vec<Property>,
}

File: typings/src/interfaces/symbol/method_map.ts
import { Declaration } from './base';
import { Property } from './property';
import { Function } from './function';

export interface MethodMap extends Declaration {
    /**
     * @brief Parent inheritance if any
     * @readonly
     */
    readonly parent?: string,

    /**
     * @brief Functions within this methodmap
     * @readonly
     */
    readonly methods: Function[],

    /**
     * @brief Properties within this methodmap
     * @readonly
     */
    readonly properties: Property[],
}
